src/Grpc.AspNetCore.Server/Internal/GrpcProtocolHelpers.cs

```diff
@@ -90,5 +93,71 @@ public static byte[] ParseBinaryHeader(string base64)
 
             return Convert.FromBase64String(decodable);
         }
+
+        internal static bool TryDecompressMessage(string compressionEncoding, byte[] messageData, out byte[] result)
+        {
+            switch (compressionEncoding)
+            {
+                case "gzip":
+                    {
+                        var output = new MemoryStream();
```

Comment:
The allocations in here are just.... Not great

Reply:
Yeah. All the compression APIs in .NET are stream based at the moment which makes it difficult to do efficiently 😐 

Compression off by default and is something you opt-in to. People can turn it on if they have large payloads and wire-size is more important that server allocations.

Also this will be revisited when we can get protobuf to use span.

Reply:
We still need to measure this allocates like crazy and I’d bet is extremely inefficient. I think we can change this code a bit to improve that but I don’t think we should merge this until we get perf numbers.